draft-ietf-mls-protocol.md

```diff
@@ -566,107 +653,66 @@ above:
          /     \
      ns[1]      F
      /  \      / \
-ns[0]    B    C   D
+    A   ns[0] C   D
 ~~~~~
 
-## Blank Nodes and Resolution
 
-A node in the tree may be _blank_, indicating that no value is
-present at that node.  The _resolution_ of a node is an ordered list
-of non-blank nodes that collectively cover all non-blank descendants
-of the node.  The nodes in a resolution are ordered according to
-their indices.
+## Synchronizing Views of the Tree
 
-* The resolution of a non-blank node is a one element list
-  containing the node itself
-* The resolution of a blank leaf node is the empty list
-* The resolution of a blank intermediate node is the result of
-  concatinating the resolution of its left child with the resolution
-  of its right child, in that order
-
-For example, consider the following tree, where the "\_" character
-represents a blank node:
-
-~~~~~
-      _
-    /   \
-   /     \
-  _       CD
- / \     / \
-A   _   C   D
-
-0 1 2 3 4 5 6
-~~~~~
-
-In this tree, we can see all three of the above rules in play:
-
-* The resolution of node 5 is the list [CD]
-* The resolution of node 2 is the empty list []
-* The resolution of node 3 is the list [A, CD]
-
-## Ratchet Tree Updates
-
-In order to update the state of the group such as adding and
-removing clients, MLS messages are used to make changes to the
-group's ratchet tree.  The member proposing an update to the
-tree transmits a set of values for intermediate nodes in the
-direct path of a leaf. Other members in the group
-can use these nodes to update their view of the tree, aligning their
-copy of the tree to the sender's.
+The members of the group need to keep their views of the tree in
+sync and up to date.  When a client proposes a change to the tree
+(e.g., to add or remove a member), it transmits a set of values for
+intermediate nodes in the direct path of a leaf. Other members of
+the group can use these nodes to update their view of the tree,
+aligning their copy of the tree to the sender's.
 
 To perform an update for a leaf, the sender transmits the following
 information for each node in the direct path of the leaf:
 
 * The public key for the node
-* Zero or more encrypted copies of the node's parent secret value
+* Zero or more encrypted copies of the path secret corresponding to
+  the node's parent
```

Comment:
Double check this with me: If I am at the leaf, I am _my_ public key, and zero encrypted copies of _my_ path secret (this is `pk(ns[0])   |   [nothing]` in the diagram)

If I am a level up, I am encrypting _my_ public key, and 1 encrypted copy of _my_ path secret (this is `pk(ns[1])   |   E(pk(A), ps[1])` in the diagram).

So this should say:

* The public key for the node
* Zero or more copies of the node's path secret, encrypted for someone in the resolution of the node's sibling

Or, since that last detail is mentioned below, just "Zero or more encrypted copies of the node's path secret"

Reply:
Correct.  The encrypted copies are for the resolution of the non-updated child, and the leaf node has no children.  And yes, the "parent" in the second bullet shouldn't be there.